### ISSUE TITLE
Light.js not working with Infura

### ISSUE THREAD

**Issue by mjdietzx** [paritytech/js-libs#51]
Hey, been playing around with Light.js a bit, It's really awesome - really speeds up our DApp and simplifies the code! running into a problem with Infura though (MetaMask web3 provider is working well)

```js
web3jsInfura = new Web3(new Web3.providers.WebsocketProvider('wss://mainnet.infura.io/ws'));
light.setProvider(web3jsInfura.currentProvider);
```

`import { makeContract, withoutLoading } from "@parity/light.js";` the failures i'm seeing below are from reading state from a contract 

ie
```js
myContract.variable$()
          .pipe(
            withoutLoading()
          )
          .subscribe(r => {
            console.log('variable', r);
          });
```

<img width="1280" alt="screen shot 2018-11-22 at 10 46 39 am" src="https://user-images.githubusercontent.com/7217256/48894930-33db5880-ee44-11e8-8646-6a30539f1a84.png">

**if i setup provider as in docs:**

`light.setProvider(
    Api.Provider.Ws('wss://mainnet.infura.io/ws')
);`

<img width="1280" alt="screen shot 2018-11-22 at 10 57 13 am" src="https://user-images.githubusercontent.com/7217256/48895516-88330800-ee45-11e8-849c-24047cfbb19c.png">

Thanks,
Mike

**Comment by amaurymartiny** [comment]
This is definitely a bug, it works if you set `light.setProvider( Api.Provider.Ws('wss://kovan.infura.io/ws') );`, i.e. only with parity nodes.

I'll investigate this and get back to you.

**Comment by amaurymartiny** [comment]
The issue should be fixed in v3.0.5
